pkg/eks/update_config_test.go

```diff
@@ -0,0 +1,166 @@
+				*describeUpdateInput = *input
+				return input.UpdateId != nil && *input.UpdateId == *describeUpdateOutput.Update.Id
+			})).Return(p.Client.MockRequestForGivenOutput(describeUpdateInput, describeUpdateOutput), describeUpdateOutput)
+		})
+
+		It("should have no logging by default", func() {
+			err = api.SetClusterConfigDefaults(cfg)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cfg.EnableLogging).To(BeEmpty())
+		})
+
+
+		It("should handle unknown facilities", func() {
+			cfg.EnableLogging = []string{"anything", "anyOtherThing"}
+
+			err = api.SetClusterConfigDefaults(cfg)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should expand [] to none", func() {
+			cfg.EnableLogging = []string{}
+
+			err = api.SetClusterConfigDefaults(cfg)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cfg.EnableLogging).To(BeEmpty())
+
+			err = ctl.UpdateClusterConfigForLogging(cfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(clusterLogging[0].Types).To(BeEmpty())
+
+			Expect(clusterLogging[1].Types).ToNot(BeEmpty())
+			Expect(clusterLogging[1].Types).To(Equal(aws.StringSlice(api.SupportedLoggingFacilities())))
+		})
+
+		It("should expand '*' to _all_", func() {
+			cfg.EnableLogging = []string{"*"}
+
+			err = api.SetClusterConfigDefaults(cfg)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cfg.EnableLogging).To(Equal(api.SupportedLoggingFacilities()))
+
+
+			err = ctl.UpdateClusterConfigForLogging(cfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(clusterLogging[0].Types).ToNot(BeEmpty())
+			Expect(clusterLogging[0].Types).To(Equal(aws.StringSlice(cfg.EnableLogging)))
+
+			Expect(clusterLogging[1].Types).To(BeEmpty())
+		})
+
+		It("should expand 'all' to _all_", func() {
+			cfg.EnableLogging = []string{"all"}
+
+			err = api.SetClusterConfigDefaults(cfg)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cfg.EnableLogging).To(Equal(api.SupportedLoggingFacilities()))
+
+			Expect(clusterLogging[0].Types).ToNot(BeEmpty())
+			Expect(clusterLogging[0].Types).To(Equal(aws.StringSlice(cfg.EnableLogging)))
+
+			Expect(clusterLogging[1].Types).To(BeEmpty())
+		})
+
+		It("should enable some logging facilities and disable others", func() {
+			cfg.EnableLogging = []string{"authenticator", "controllerManager"}
+
+			err = api.SetClusterConfigDefaults(cfg)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cfg.EnableLogging).To(Equal([]string{"authenticator", "controllerManager"}))
+
+			err = ctl.UpdateClusterConfigForLogging(cfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(clusterLogging[0].Types).ToNot(BeEmpty())
+			Expect(clusterLogging[0].Types).To(Equal(aws.StringSlice(cfg.EnableLogging)))
+
+			Expect(clusterLogging[1].Types).ToNot(BeEmpty())
+			Expect(clusterLogging[1].Types).To(Equal(aws.StringSlice([]string{"api", "audit", "scheduler"})))
+		})
+
+		It("should enable some logging facilities and disable others", func() {
```

Comment:
I'm not sure what value is added by this test.

Reply:
Yeah, I'll remove it. It was a lazy fuzz test. We should do proper fuzz tests instead.